shardingsphere-features/shardingsphere-shadow/shardingsphere-shadow-core/src/test/java/org/apache/shardingsphere/shadow/route/future/engine/determiner/algorithm/NoteShadowAlgorithmDeterminerTest.java

```diff
@@ -0,0 +1,117 @@
+    }
+
+    private NoteShadowAlgorithm createNoteShadowAlgorithm() {
+        NoteShadowAlgorithm<String> noteShadowAlgorithm = new SimpleSQLNoteShadowAlgorithm();
+        noteShadowAlgorithm.setProps(createProps());
+        noteShadowAlgorithm.init();
+        return noteShadowAlgorithm;
+    }
+
+    private Properties createProps() {
+        Properties properties = new Properties();
+        properties.setProperty("shadow", "true");
+        properties.setProperty("user_id", "1");
+        return properties;
+    }
+
+    @Test
+    public void assertIsShadow() {
+        assertThat(shadowAlgorithmDeterminer.isShadow(createShadowDetermineCondition(), new ShadowRule(createAlgorithmProvidedShadowRuleConfiguration()), "t_order"), is(true));
+    }
+
+    private AlgorithmProvidedShadowRuleConfiguration createAlgorithmProvidedShadowRuleConfiguration() {
+        AlgorithmProvidedShadowRuleConfiguration result = new AlgorithmProvidedShadowRuleConfiguration("shadow", Arrays.asList("ds", "ds1"), Arrays.asList("shadow_ds", "shadow_ds1"));
+        result.setEnable(true);
+        result.setDataSources(createDataSources());
+        result.setTables(createTables());
+        result.setShadowAlgorithms(createShadowAlgorithms());
+        return result;
+    }
+
+    private Map<String, ShadowAlgorithm> createShadowAlgorithms() {
+        Map<String, ShadowAlgorithm> result = new LinkedHashMap<>();
+        result.put("user_id-insert-regex-algorithm", createNoteShadowAlgorithm());
+        return result;
+    }
+
+    private Map<String, ShadowTableConfiguration> createTables() {
+        Map<String, ShadowTableConfiguration> result = new LinkedHashMap<>();
+        result.put("t_order", new ShadowTableConfiguration(createShadowAlgorithmNames()));
+        return result;
+    }
+
+    private Collection<String> createShadowAlgorithmNames() {
+        Collection<String> result = new LinkedList<>();
+        result.add("user_id-insert-regex-algorithm");
+        return result;
+    }
+
+    private Map<String, ShadowDataSourceConfiguration> createDataSources() {
+        Map<String, ShadowDataSourceConfiguration> result = new LinkedHashMap<>();
+        result.put("ds-data-source", new ShadowDataSourceConfiguration("ds", "ds_shadow"));
+        result.put("ds1-data-source", new ShadowDataSourceConfiguration("ds1", "ds1_shadow"));
+        return result;
+    }
+
+    private ShadowDetermineCondition createShadowDetermineCondition() {
+        ShadowDetermineCondition shadowDetermineCondition = new ShadowDetermineCondition(ShadowOperationType.INSERT);
+        shadowDetermineCondition.initSqlNotes(createSqlNotes());
+        return shadowDetermineCondition;
+    }
+
+    private Collection<String> createSqlNotes() {
+        Collection<String> result = new LinkedList<>();
+        result.add("/*user_id=1,shadow=true*/");
+        return result;
+    }
+
```

Comment:
@LeeGuoPing Thank you for your contribution, please delete the extra blank lines.

Reply:
> @LeeGuoPing Thank you for your contribution, please delete the extra blank lines.

@soulasuna I have deleted it just now.